pymeasure/instruments/formfactor/velox.py

```diff
@@ -0,0 +1,270 @@
+            - item[3]: Total number of subdies
+
+        """
+
+        coordinates = ""
+        for arg in args:
+            if type(arg) in [list, tuple]:
+                coordinates += " ".join(map(str, arg))
+            else:
+                coordinates += str(arg).replace(",", " ").replace(";", " ")
+            coordinates += " "
+
+        if len(coordinates):
+            coordinates = " ".join(coordinates.split())  # ensure single space between elements
+
+        command = f"StepNextDie {coordinates}".strip()
+        return self.values(command,
+                           separator=" ",
+                           cast=int,
+                           )
+
+    enabled = Channel.measurement(
+        "IsAppRegistered WaferMap",
+        """Get whether the WaferMap module is loaded or not.""",
+        map_values=True,
+        values={True: 1, False: 0}
+        )
+
+
+class Velox(SCPIMixin, Instrument):
+    """A class representing the FormFactor Velox wafer prober.
+
+    Example for a measurement of an entire wafer:
+
+    .. code-block:: python
+
+        from pymeasure.instruments.formfactor.velox import Velox
+
+        prober = Velox("GPIB0::28::INSTR")  # replace the resource to your needs
+        coordinates = prober.wafermap.step_firstdie()  # move to the first die in the wafermap
+        while not prober.error_code:
+            prober.chuck.move_contact()  # move to contact height
+            # ...
+            # code for measurement, processing and storing data
+            # ...
+            prober.chuck.move_separation()  # move to separation height
+            coordinates = prober.wafermap.step_nextdie()  # move to the next die in the wafermap
+
+    """
+
+    error_code = 0
+    error_message = ""
+
+    chuck = Instrument.ChannelCreator(Chuck)
+    wafermap = Instrument.ChannelCreator(WaferMap)
+
+    def __init__(self, adapter, name="FormFactor Velox",
+                 timeout=40000,
+                 **kwargs):
+        super().__init__(
+            adapter, name,
+            timeout=timeout,
+            **kwargs
+        )
+
+    version = Instrument.measurement(
+        "ReportSoftwareVersion",
+        """Get the main software version of Velox (str).""",
+        )
+
+    def read(self):
+        """
+        Read the response and check for errors.
+
+        :raise: *VeloxError* if an error is detected.
+
+        """
+
+        self.error_code = 0
+        self.error_message = ""
+
+        got = super().read()
+
+        # some commands like *IDN do not return '0: ...'
+        # in such a case return the entire response string
+        if ":" not in got:
+            return got
+
+        got = got.partition(":")
+        self.error_code = int(got[0])
+        response = got[2].strip()
+
+        if self.error_code:
+            self.error_message = response
+            if self.error_code not in EXPECTED_ERRORS:
+                raise VeloxError(f"{self.error_code}: {self.error_message}")
```

Comment:
I typically prefer to use builtin exceptions as you do not have to import the exception as well.